### PR TITLE
feat(design-system): expose UI components via Storybook

### PR DESCRIPTION
**Pull request by thlmenezes**
## Description of changes

- Exposes UI components via Storybook, including:
    - Devopness theme configuration: colors, fonts and icons
    - Uses latest components dependencies: [Storybook](https://storybook.js.org/) 8, [Styled Components](https://styled-components.com/) 6, [React](https://react.dev/) 18, [MUI](https://mui.com/material-ui/) 6

## GitHub issues resolved by this PR

N/A

## Quality Assurance

- Once the changes in this PR are merged and deployed, success criteria is: Expose UI components via Storybook

## More info

[DVN-16185]

[DVN-16185]: https://devopness.atlassian.net/browse/DVN-16185?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ui/react/.gitignore [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+!.vscode/extensions.json
+.idea
+.DS_Store
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
+
+*storybook.log
```
Comment (jfoliveira) [inline packages/ui/react/.gitignore:26]: NIT-pick: Final new line? 👁️ 


image

**Review by jfoliveira — CHANGES_REQUESTED** [review]

File: packages/ui/react/src/components/Templates/Tabs/Tabs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import type { Meta, StoryObj } from '@storybook/react'
```
Comment (jfoliveira) [inline packages/ui/react/src/components/Templates/Tabs/Tabs.stories.tsx:1]: @thlmenezes could you please ensure a minimal component description for each component?

For example, when accessing `Button > Docs` we see the text 
```
Primary UI component for user interaction
```

However, when accessing Primitives, Templates, Tooltips, Tab, etc, I couldn't understand their purpose.

I believe a short text in each componente would be helpful.

image

File: packages/ui/react/.gitignore (hunk 1 above)
Comment (thlmenezes) [reply packages/ui/react/.gitignore:26]: Just to be certain, is this your suggestion?

```suggestion
*storybook.log

```

Comment (jfoliveira) [reply packages/ui/react/.gitignore:26]: Yes. All files should end with a new line (CR + LF) as indicated by GitHub with a red icon ⛔ 
For context, this discussion might be of your interest https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: packages/ui/react/src/components/Templates/Tabs/Tabs.stories.tsx (hunk 2 above)
Comment (thlmenezes) [reply packages/ui/react/src/components/Templates/Tabs/Tabs.stories.tsx:1]: > I believe a short text in each componente would be helpful.

Agreed, added some texts to describe both components and a new deployment is on the way https://app.devopness.com/actions/40235

Comment (thlmenezes) [reply packages/ui/react/src/components/Templates/Tabs/Tabs.stories.tsx:1]: image

image

**Review by jfoliveira — CHANGES_REQUESTED** [review]

File: packages/ui/react/.storybook/preview.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        color: /(background|color)$/i,
+        date: /Date$/i,
+      },
+    },
+  },
+  tags: [
+    'autodocs',
+  ],
+}
+
```
[the remark this answers is not in the document]
Comment (thlmenezes) [reply packages/ui/react/.storybook/preview.ts:17]: Done, also included other README files with related documentation links and information about its folder.

Could you review it again?

**Review by jfoliveira — APPROVED** [review]

**Review by souz4-my — APPROVED** [review]

**Pull request merged**